### PR TITLE
Improvements on the network graph

### LINKED ISSUE #59: Network graph improvements
I am opening this to keep some notes on ideas for future improvement of the network graph functionality.

* Grouping of nodes belonging to the same token network: [Grouping nodes in a Force-Directed Graph](https://bl.ocks.org/XavierGimenez/a8e8c5e9aed71ba96bd52332682c0399)
* Highlight neighbor connections when user presses a node: [Interactive & Dynamic Force-Directed Graphs with D3](https://medium.com/ninjaconcept/interactive-dynamic-force-directed-graphs-with-d3-da720c6d7811)

### LINKED ISSUE #60: Visualization improvements
Some ideas for further improving the visualization:
- [ ] Don't let the nodes run out of the visualization area
- [x] Add better tooltips (show them faster, show more infos)
- [x] Make nodes selectable and when selected highlight channels

### PR DESCRIPTION
**Pull request by kelsos**
Closes #60 
Closes #59

### CODE REVIEW

File: frontend/src/app/components/network-graph/network-graph.component.ts [hunk 1]
```diff
@@ -17,6 +18,10 @@ interface SimulationLink extends SimulationLinkDatum<SimulationNode>, Link {
   styleUrls: ['./network-graph.component.css']
 })
 export class NetworkGraphComponent implements OnInit, OnChanges {
+  private static HIGHLIGHT_COLOR = '#2E41FF';
```
Comment (palango) [inline frontend/src/app/components/network-graph/network-graph.component.ts]: Can we put this in some CSS somewhere?

File: frontend/src/app/components/network-graph/network-graph.component.ts [hunk 2]
```diff
@@ -175,27 +186,29 @@ export class NetworkGraphComponent implements OnInit, OnChanges {
...
       .attr('class', 'links')
-      .attr('stroke-width', '2')
-      .attr('stroke-opacity', 0.6)
       .selectAll('.link');
 
     this.node = this.svg.append('g')
       .attr('class', 'nodes')
-      .attr('fill', '#1A237E')
       .attr('stroke-width', '1')
       .attr('stroke', '#fff')
```
Comment (palango) [inline frontend/src/app/components/network-graph/network-graph.component.ts]: Same here, would be nice to make this easier to configure/change.

File: frontend/src/app/components/network-graph/network-graph.component.ts [hunk 3]
```diff
@@ -221,15 +236,25 @@ export class NetworkGraphComponent implements OnInit, OnChanges {
...
         .on('drag', this.dragged)
         .on('end', datum => this.dragended(datum, this.simulation)))
       .merge(nodes);
 
-    node.append('title').text(d => d.id);
+    const tooltip = (d: Node) => d.id + '\n\n' +
+      'Token: ' + d.tokenAddress + '\n\n' +
+      'Open Channels: ' + d.openChannels + '\n' +
+      'Closed Channels: ' + d.closedChannels + '\n' +
+      'Settle Channels: ' + d.settledChannels;
```
Comment (palango) [inline frontend/src/app/components/network-graph/network-graph.component.ts]: s/Settle/Settled

File: frontend/src/app/components/network-graph/network-graph.component.ts (hunk 1 above)
Comment (kelsos) [reply frontend/src/app/components/network-graph/network-graph.component.ts]: Would it make a difference? These colors are only internal in the component for coloring the nodes, the are not used anywhere else. We still have only one place to change. 

Moving them to css would make the logic more complex. 
https://github.com/raiden-network/explorer/pull/62/files#diff-a66689adaf934d8c6025c371e5381f41R324

Currently the `fill` changes depending on the state of the node (normal,highlighted, or selected). If we move the colors from the component to css we will have to add a `class` `attribute` for highlighted or selected that have static color values, and remove the `class` use a fill for the programmatically generated colors for the nodes in any other case.

**Pull request merged**